sktime/utils/_testing/deep_equals.py

```diff
@@ -0,0 +1,131 @@
+# -*- coding: utf-8 -*-
+"""
+Testing utility to compare equality in value for nested objects
+
+Objects compared can have one of the following valid types:
+    types compatible with != comparison
+    pd.Series, pd.DataFrame, np.ndarray
+    lists, tuples, or dicts of a valid type (recursive)
+"""
+
+__author__ = ["fkiraly"]
+
+__all__ = ["deep_equals"]
+
+
+import numpy as np
+import pandas as pd
+
+
+def deep_equals(x, y):
```

Comment:
as before, I still think this is overcomplicated and the costs here outweigh the benefits of having arbitrary tag values, for boolean and string tags there exist well-tested inbuilt comparison functionality

Reply:
removed, resolved.